### PR TITLE
Upgrading to storybook 4.0.0

### PR DESCRIPTION
**Pull request by nickstaroba**
Also handles the bug noted in #10

### CODE REVIEW

File: src/register.js [hunk 1]
```diff
@@ -63,18 +63,19 @@ class StatePanel extends React.Component {
   };
 
   render() {
+    const { active } = this.props;
```
Comment (dump247) [inline src/register.js:67]: Please add `active: T.bool` to the propTypes.

**Comment by dump247** [comment]
Is this change backward compatible? I haven't had the chance to test with storybook < 4, but it doesn't look like it is compatible. If it is not, then it should be changed to be backward compatible. I _think_ it would be backward compatible if the render function is ` (opts) => <StatePanel channel={channel} api={api} active={!opts || opts.active} />`. I don't know if storybook < 4 passes any parameter to the render function, so this might not work.

An alternative would be a major version change. That would also be a good time to switch to storybook 4 [decorator parameters](https://storybook.js.org/addons/api/#makedecorator-api) and get rid of the legacy function formats.

**Comment by nickstaroba** [comment]
> Is this change backward compatible? 

From my attempts, no, but I won't say that definitively until I've had more time to dive into storybook. 

> ` (opts) => <StatePanel channel={channel} api={api} active={!opts || opts.active} />`.

Did not work in my attempt. 

> An alternative would be a major version change. That would also be a good time to switch to storybook 4 [decorator parameters](https://storybook.js.org/addons/api/#makedecorator-api) and get rid of the legacy function formats.

I can look into this more over the weekend. I'm still very new to storybook.

File: src/register.js (hunk 1 above)
Comment (nickstaroba) [reply src/register.js:67]: done

**Comment by chengcyber** [comment]
Hi @dump247. IMO, it's better to bump the major version because it's not backward compatible. Shall we merge this pr and bump the major version first? thus, we can do more optimization upon the new version.

**Comment by jelche** [comment]
What is the consensus? v2.0.0 bump?

**Comment by ozio** [comment]
Any news?

**Comment by OmriAharon** [comment]
Any chance to go ahead and merge this with a major bump? This is a blocker for most of us to update to v4.

**Comment by dump247** [comment]
Should be resolved in #17

**Pull request closed without merging**